### ISSUE TITLE
"All day" events start and end at 01:00 instead of 00:00

### ISSUE THREAD

**Issue by logological** [mguessan/davmail#351]
I'm using DavMail 6.2.2 with Thunderbird 115 on openSUSE Tumbleweed. My system's timezone is set to America/Winnipeg (which is Central Time), and Thunderbird is configured to use this system timezone.

If I create a new event in Thunderbird and check the "All day Event" box, then I expect it to begin at 00:00 on the start date and to end at 24:00 on the end date.  However, for DavMail calendars only, Thunderbird shows these events as beginning at 01:00 on the start date and ending at 01:00 on the day following the end date.

Perhaps this problem is related to Issue 342.

**Comment by logological** [comment]
Interestingly, if I edit the all-day event once it's been created, I see that the "All day Event" box is unchecked, and that start and end times of 01:00 are explicitly given in the start and end date fields.  If I re-check the "All day Event" box, correct the end date, and re-save the event, then this time the event is saved with the correct starting and ending times.  So it seems it's only when all-day events are _initially_ created that this problem arises.

**Comment by mguessan** [comment]
We had issues on allday events a long time ago in WebDav mode, guess it's not the case here, need to investigate further the logs to understand why th allday flag is lost on create

**Comment by mguessan** [comment]
Reviewing older tickets, is this still an issue with latest release?

Just a hint on troubleshooting: all day events are marked with a microsoft only CDO ALLDAY property, and on thunderbird side they have a date in YYYYMMDD format without time.

**Comment by logological** [comment]
Yes, I can still reproduce the problem when Thunderbird's time zone is set to America/Winnipeg.  (I am currently using Thunderbird 150.0.1 and DavMail 6.7.0.)  The problem does not occur if I change Thunderbird's time zone to America/Chicago, even though both time zones are on Central Daylight Time.

**Comment by mguessan** [comment]
Interesting, I don't reproduce this, at least on trunk:

When creating an allday event the timezone is not sent, event dates have no time information and DavMail adds the CDO flag:

```
DTSTART;VALUE=DATE:20260529
DTEND;VALUE=DATE:20260530
X-MICROSOFT-CDO-ALLDAYEVENT:TRUE
```

Can you please check DavMail logs on create?
